.github/workflows/ci.yml

```diff
@@ -189,6 +189,21 @@ jobs:
       shell: bash
       run: ${{ env.CARGO }} test --bin rg ${{ env.TARGET_FLAGS }} flags::defs::tests::available_shorts -- --nocapture
 
+     # Setup and compile on the wasm32-wasi target
+  wasm:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: stable
+    - name: Add wasm32-wasi target
+      run: rustup target add wasm32-wasi
+    - name: Basic build
+      run: cargo build --verbose
```

Comment:
Did running tests via `wasmtime` fail? If so this is fine, but if they worked, then we should just include them here.

Reply:
Indeed, running tests via `wasmtime` does fail (but compilation works):
```
error[E0433]: failed to resolve: could not find `unix` in `os`
   --> tests/util.rs:197:22
    |
197 |         use std::os::unix::fs::symlink;
    |                      ^^^^ could not find `unix` in `os`
    |
```

Reply:
Ah fair enough. We can save that for another day.